### ISSUE TITLE
Seamonkey extensions get incorrect maxVersion.

### ISSUE THREAD

**Issue by musically-ut** [mozilla-jetpack/jpm#518]
While porting [Lovely forks](https://github.com/musically-ut/lovely-forks) to SeaMonkey, I ran into this rather arcane issue.

The current release used the version `43` as the maximum version, which has [changed in HEAD to `*`](https://github.com/mozilla-jetpack/jpm/commit/a8606061d7850a3ab5def35238d28eb586b511cc).

However, [SeaMonkey's valid application versions](https://addons.mozilla.org/en-US/firefox/pages/appversions/), for some odd reason, do not have `*` as a valid option.

Hence, the `install.rdf` generated with only a `minVersion` specified for SeaMonkey is deemed invalid when uploaded to Addons site because:

  - (With the release `1.0.6`): `43` is an invalid max version for SeaMonkey
  - (With HEAD): `*` is an invalid max version for SeaMonkey.

I am not sure what is the best way to resolve the issue. Perhaps having an GUID/engine aware `MAX_VERION` with the `*` fallback for unknown platforms would be the best solution?

Alternatively, what is the best channel to contact Mozilla to allow `*` for SeaMonkey as well?

**Comment by asamuzaK** [comment]
Even if `*` is implemented, default MIN_VERSION (38.0a1) will still be invalid for SeaMonkey.
Isn't it?

So, I think it's better to add "engines" field in package.json
`"engines": {
  "firefox": ">=38.0a1 <=*",
  "seamonkey": ">=2.0a <=2.40"
}`

[package.json - Mozilla | MDN](https://developer.mozilla.org/en-US/Add-ons/SDK/Tools/package_json)

**Comment by musically-ut** [comment]
Yes, that is what I had done. However, that does mean that I have to keep releasing new version of my addon as newer versions of SeaMonkey are released. It should not be needed since I am using stable SDK API. That, in my opinion, should be the desired behavior if the `MAX_VERSION` is omitted.

However, if I omit specifying the `MAX_VERSION`, then `jpm` would substitute it by `*`, irrespective of whether it is a valid version for the `engine` or not. I think that is a bug. 
 
Alternatively, it may be easier to just add `*` as a valid version for SeaMonkey: https://github.com/mozilla/addons/issues/74

**Comment by freaktechnik** [comment]
Doesn't SeaMonkey ignore the max version by default too? So it's just AMO that actually looks at it (and decides to show an extension based on it or not, I think).
